### PR TITLE
add firewall to prometheus 

### PR DESCRIPTION
**Pull request by bouriquet**

### CODE REVIEW

File: roles/addons/prometheus_server/tasks/main.yml
Comment (btravouillon) [inline roles/addons/prometheus_server/tasks/main.yml:44]: Can you use prometheus service instead of 9090/tcp?

```
# firewall-offline-cmd --info-service prometheus
prometheus
  ports: 9090/tcp
  protocols: 
  source-ports: 
  modules: 
  destination: 
  includes: 
  helpers: 
```
It makes the firewall configuration easier to read with `firewall-cmd --info-zone <zone>`.

File: roles/addons/prometheus_server/tasks/main.yml
Comment (btravouillon) [inline roles/addons/prometheus_server/tasks/main.yml:45]: This should not work: `service: 9093/tcp`.
Is it alertmanager? There is no pre-defined service in EL8 (at least I can't find one). If possible, can you create the service in a previous task? (see [this example gist](https://gist.github.com/btravouillon/d196488211d69d2cbea22c44ae268410))

File: roles/addons/prometheus_server/tasks/main.yml
Comment (btravouillon) [inline roles/addons/prometheus_server/tasks/main.yml:46]: `9100/tcp` might be the port for node_exporter. I expect node_exporter to be installed with the role prometheus_client (which means there is a defect in the role, not in this PR). Am I wrong?

**Review by btravouillon — CHANGES_REQUESTED** [review]
Mandatory comment to request changes: please see my previous comments.

File: roles/addons/prometheus_server/tasks/main.yml
Comment (bouriquet) [reply roles/addons/prometheus_server/tasks/main.yml:45]: I took thoses ports from prometheus wiki : https://github.com/prometheus/prometheus/wiki/Default-port-allocations

File: roles/addons/prometheus_server/tasks/main.yml
Comment (bouriquet) [reply roles/addons/prometheus_server/tasks/main.yml]: Right, replaced by prometheus service

File: roles/addons/prometheus_server/tasks/main.yml
Comment (bouriquet) [reply roles/addons/prometheus_server/tasks/main.yml]: 9100 removed from the list, already opened by prometheus client role

File: roles/addons/prometheus_server/tasks/main.yml
Comment (btravouillon) [inline roles/addons/prometheus_server/tasks/main.yml:36]: Please prefer /etc/firewalld/services/alertmanager.xml. (https://firewalld.org/documentation/configuration/directories.html)

Comment (bouriquet) [reply roles/addons/prometheus_server/tasks/main.yml]: I took your snippet, didnt knew about /etc/firewalld, thanks.

Comment (btravouillon) [reply roles/addons/prometheus_server/tasks/main.yml]: You did fast! I updated the snippet one minute after posting when I saw my mistake 😄

**Comment by btravouillon** [comment]
Did you test the code?

**Comment by bouriquet** [comment]
> Did you test the code?

Removed the manually added service before testing this time, firewall reload need to be added after service declaration.

**Review by oxedions — APPROVED** [review]
Dear @bouriquet 

Thank you for this PR.

After a look, I found it ok.
Waiting for @btravouillon approval and merging.

**Comment by btravouillon** [comment]
I still have issues with prometheus_client.

```
[WARNING]: While constructing a mapping from /etc/bluebanquise/roles/addons/prometheus_client/tasks/main.yml,
 line 8, column 3, found a duplicate dict key (when). Using last defined value only.
```

With this inventory, the task is always skipped:
```
    monitoring:
      exporters:
        node_exporter:
        package: node_exporter
        service: node_exporter
        port: 9100

# ansible-playbook playbooks/monitoring.yml --tags prometheus_client -e ep_firewall=true --limit management1
[...]
TASK [prometheus_client : firewalld █ Add services to firewall's public zone] ********************************************************************************************************************
Friday 11 September 2020  09:03:33 -0400 (0:00:00.142)       0:00:02.350 ******
skipping: [management1] => (item=None) 
skipping: [management1] => (item=node_exporter) 
skipping: [management1] => (item=node_exporter) 
skipping: [management1] => (item=9100) 
```

Reading the code, item.port is not defined.

**Comment by bouriquet** [comment]
> I still have issues with prometheus_client.
> 
> ```
> [WARNING]: While constructing a mapping from /etc/bluebanquise/roles/addons/prometheus_client/tasks/main.yml,
>  line 8, column 3, found a duplicate dict key (when). Using last defined value only.
> ```
> 
> With this inventory, the task is always skipped:
> 
> ```
>     monitoring:
>       exporters:
>         node_exporter:
>         package: node_exporter
>         service: node_exporter
>         port: 9100
> 
> # ansible-playbook playbooks/monitoring.yml --tags prometheus_client -e ep_firewall=true --limit management1
> [...]
> TASK [prometheus_client : firewalld █ Add services to firewall's public zone] **************************
> Friday 11 September 2020  09:03:33 -0400 (0:00:00.142)       0:00:02.350 ******
> skipping: [management1] => (item=None) 
> skipping: [management1] => (item=node_exporter) 
> skipping: [management1] => (item=node_exporter) 
> skipping: [management1] => (item=9100) 
> ```
> 
> Reading the code, item.port is not defined.

Are you sure about your identation ? i tried with the example from README and it doesnt have the same indentation
(Works well with this example)

monitoring:
  exporters:
    node_exporter:
      package: node_exporter
      service: node_exporter
      port: 9100

**Comment by btravouillon** [comment]
Indeed, bad copy&paste... my bad.

**Comment by btravouillon** [comment]
However the test still fails.

```
# ansible-playbook playbooks/monitoring.yml --tags prometheus_client -e ep_firewall=false --limit management1
[WARNING]: While constructing a mapping from /etc/bluebanquise/roles/addons/prometheus_client/tasks/main.yml,
  line 8, column 3, found a duplicate dict key (when). Using last defined value only.
[...]
TASK [prometheus_client : firewalld █ Add services to firewall's public zone] ***************
Friday 11 September 2020  11:05:36 -0400 (0:00:01.618)       0:00:03.791 ****** 
changed: [management1] => (item={'package': 'node_exporter', 'service': 'node_exporter', 'port': 9100})
```

**Review by btravouillon — APPROVED** [review]
Thanks for this PR!

**Pull request merged**
